kokoro/gcp_ubuntu/releases/release_utils.sh

```diff
@@ -0,0 +1,162 @@
+# makes HTTP requests and returns a status code
+# Globals:
+#   None
+# Arguments:
+#   A non optional HTTP header, a url
+# Outputs:
+#   A http status code
+#######################################
+httpGetErrorCode() {
+  http_header="$1"
+  http_url="$2"
+
+  http_response=$(curl -s -w "%{http_code}" -H ${http_header} ${http_url})
+
+  echo $(tail -n1 <<< "$http_response")  # get the last line as error code
+}
+
+#######################################
+# makes HTTP post requests, expects given status code
+# Globals:
+#   None
+# Arguments:
+#   Expected status code
+#   A non optional HTTP header
+#   A url
+#   Content to POST
+# Outputs:
+#   A http status code
+#######################################
+httpPostCheckStatus() {
+  expected_http_status="$1"
+  http_header="$2"
+  http_url="$3"
+  post_data="$4"
+  
+  http_response=$(curl -s -w "%{http_code}" -X POST -H ${http_header} ${http_url} -d "${post_data}" )
+
+  http_response_code=$(tail -n1 <<< "$http_response")  # get the last line
+  http_response_content=$(sed '$ d' <<< "$http_response") # get all but the last line which contains the status code
+  
+  if [ $http_response_code !=  $expected_http_status ]; then
+    err "ERROR! Http request failed with code ${http_response_code}," \
+      "response content is ${http_response_content}"   # write error message to stderr
+  fi
+  echo ${http_response_content:3} # remove trailing code before returning
+}
+
+#######################################
+# makes HTTP patch requests, expects given status code
+# Globals:
+#   None
+# Arguments:
+#   Expected status code
+#   A non optional HTTP header
+#   A url
+#   Content to POST
+# Outputs:
+#   A http status code
+#######################################
+httpPatchCheckStatus() {
+  expected_http_status="$1"
+  http_header="$2"
+  http_url="$3"
+  post_data="$4"
+
+  http_response=$(curl -s -w "%{http_code}" -X PATCH -H ${http_header} ${http_url} -d ${post_data} )
+
+  http_response_code=$(tail -n1 <<< "$http_response")  # get the last line
+  http_response_content=$(sed '$ d' <<< "$http_response") # get all but the last line which contains the status code
+
+  if [ $http_response_code !=  $expected_http_status ]; then
+    err "ERROR! Http request failed with code ${http_response_code}," \
+      "response content is ${http_response_content}"   # write error message to stderr
+  fi
+  echo ${http_response_content:3} # remove trailing code before returning
+}
+
+#######################################
+# makes HTTP GET requests, expects given status code
+# Globals:
+#   None
+# Arguments:
+#   Expected status code
+#   A non optional HTTP header
+#   A url
+# Outputs:
+#   A http status code
+#######################################
+httpGetCheckStatus() {
+  expected_http_status="$1"
+  http_header="$2"
+  http_url="$3"
+
+  http_response=$(curl -w "%{http_code}" -H ${http_header} ${http_url}  )
+
+  http_response_code=$(tail -n1 <<< "$http_response")  # get the last line
```

Comment:
lines 139-145 are exactly the same for `httpGetCheckStatus`, `httpPatchCheckStatus` and `httpPostCheckStatus`. Consider moving them to a function